dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentFactoryImpl.java

```diff
@@ -1981,35 +1907,223 @@ public Contentlet save(Contentlet contentlet) throws DotDataException, DotStateE
+            int j = 0;
+            for(final String wysiwygField : wysiwygFields ) {
+                wysiwyg.append(wysiwygField);
+                j++;
+                if( j < wysiwygFields.size() ) wysiwyg.append(",");
+            }
+            upsertValues.add(wysiwyg.toString());
+        } else{
+            upsertValues.add(null);
+        }
+    }
+
+    /**
+     * Add dates, text, text_area and integer fields from the contentlet's fields map to the {@link DotConnect} object
+     * @param upsertValues
+     * @param fieldsMap
+     * @param prefix
+     */
+    private void addDynamicFields(final List<Object> upsertValues, final Map<String, Object> fieldsMap, final String prefix)
+            throws JsonProcessingException {
+        Object defaultValue = null;
+
+        if (prefix.equals("integer") || prefix.equals("float")){
+            defaultValue = 0;
+        } else if (prefix.equals("bool")){
+            defaultValue = Boolean.FALSE;
+        }
+
+        for (int i = 1; i <= 25; i++) {
+            if (fieldsMap.containsKey(prefix + i)) {
+
+                if (prefix.equals("date") && UtilMethods.isSet(fieldsMap.get(prefix + i))){
+                    upsertValues.add(new Timestamp(((Date) fieldsMap.get(prefix + i)).getTime()));
+                } else{
+                    if (prefix.startsWith("text") && fieldsMap.get(prefix + i) instanceof Map){
+                        upsertValues.add(mapper.writeValueAsString(fieldsMap.get(prefix + i)));
+                    }else{
+                        upsertValues.add(fieldsMap.get(prefix + i));
+                    }
+                }
+
+            } else {
+                upsertValues.add(defaultValue);
+            }
+        }
+    }
+
+    private Map<String, Object> getFieldsMap(Contentlet contentlet) throws DotDataException {
+        final Map<String, Object> fieldsMap = new HashMap<>();
+        final List<Field> fields = FieldsCache.getFieldsByStructureInode(contentlet.getContentTypeId());
+        for (Field field : fields) {
+            if (field.getFieldType().equals(Field.FieldType.HOST_OR_FOLDER.toString())) {
+                continue;
+            }
+            if (field.getFieldType().equals(Field.FieldType.BINARY.toString())) {
+                continue;
+            }
+
+            if(!APILocator.getFieldAPI().valueSettable(field)){
+                continue;
+            }
+            Object value;
+            value = contentlet.get(field.getVelocityVarName());
+
+            try {
+                if(value != null && value instanceof Timestamp){
+                    value = new Date(((Timestamp)value).getTime());
+                }
+
+                fieldsMap.put(field.getFieldContentlet(), value);
+            } catch (final IllegalArgumentException e) {
+                throw new DotDataException("Unable to set field value",e);
+            }
+        }
+
+        return fieldsMap;
+    }
```

Comment:
it's a nitpick. but I like returning an immutable list. If they are reused to add more elements I like  to see a new list created

Reply:
Done

Reply:
Good